test/SILOptimizer/OSLogPrototypeCompileDiagnostics.swift

```diff
@@ -0,0 +1,44 @@
+// RUN: %target-swift-frontend -swift-version 5 -emit-sil -primary-file %s -o /dev/null -verify
+// REQUIRES: OS=macosx || OS=ios || OS=tvos || OS=watchos
+
+// Tests for the diagnostics produced by the OSLogOptimization pass that
+// performs compile-time analysis and optimization of the new os log prototype
+// APIs. The tests here check whether bad user inputs are diagnosed correctly.
+// The tests here model possible invalid inputs to the os log methods.
+// TODO: the diagnostics will be improved.
+
+import OSLogPrototype
+
+if #available(OSX 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *) {
+
+  func testDynamicLogMessage(h: Logger, message: OSLogMessage) {
+    h.log(level: .debug, message) // expected-error {{os log methods must be passed a string interpolation literal. 'OSLogMessage' must not be constructed explicitly}}
+  }
+
+  func testNonconstantFormatOption(h: Logger, formatOpt: IntFormat) {
+    h.log(level: .debug, "Minimum integer value: \(Int.min, format: formatOpt)")
+    // expected-error @-1 {{'OSLogInterpolation.formatString' is not a constant: formatting and privacy options must be literals}}
+  }
+
+  func testNonconstantPrivacyOption(h: Logger,  privacyOpt: Privacy) {
+    h.log(level: .debug, "Minimum integer value: \(Int.min, privacy: privacyOpt)")
+    // expected-error @-1 {{'OSLogInterpolation.formatString' is not a constant: formatting and privacy options must be literals}}
+  }
+
+  // FIXME: the following two tests should produce diagnostics and are not
+  // valid uses of the log APIs. Only string interpolation literal should be
+  // passed to log methods.
```

Comment:
Yes, but actually, this case should even produce a compile-time error. The error should be on "h.log(level: .debug, formatString)", which expects  the second parameter to be a string interpolation.  `let formatString = ...` is not by itself an error, it will not be optimized as the optimization is driven by the calls to the `log` function.